### PR TITLE
Updates to make VSCode work by default with Metals.

### PR DESCRIPTION
**Pull request by clouser2**
### Description
**bin/vscode-setup**
*   Updated the Maven properties indicating that this is Java 11. Fixes a host of problems reported by VSCode around our use of lambdas and other language features. If unspecified, this defaults to Java 5.

**civiform.code-workspace** 
*   Inverted the order of project definitions. Prior to this, Metals didn't detect a SBT and didn't automatically compile.
*   Added .bloop / target files to list of exemptions for searching. This prevents these files from showing in the quick file open candidate list.

**server/build.sbt**
*   Marked auto-value-parent as [pomOnly](https://stackoverflow.com/q/45608473) since it doesn't have a corresponding JAR. Without this, the call to makePom from vscode-setup adds this as a dependency and VSCode fails to resolve it, preventing any further indexing from occurring.

With these changes, the remaining problems reported by VSCode are largely around AutoValue and auto-generated Scala routes. We'll hopefully address these separately.

### Tested
*   Created a fresh clone of the repository and cleared all home directory Maven / Ivy caches. 
*   Ran `bin/vscode-setup`
*   Opened `civiform.code-workspace` from VSCode and waited for build / index to complete
*   Observed auto-complete and jump to definition working as intended  
*   Ran `bin/run-dev` and confirmed I could make requests
*   Confirmed `bin/build-and-push` successfully built an image

### CODE REVIEW

File: server/build.sbt [hunk 1]
```diff
@@ -67,7 +67,7 @@ lazy val root = (project in file("."))
       // Autovalue
       "com.google.auto.value" % "auto-value-annotations" % "1.9",
       "com.google.auto.value" % "auto-value" % "1.9",
-      "com.google.auto.value" % "auto-value-parent" % "1.9",
+      "com.google.auto.value" % "auto-value-parent" % "1.9" pomOnly(),
```
[the remark this answers is not in the document]
Comment (kmcnellis) [reply server/build.sbt:70]: And alternate approach would be to use the script to modify this line in the generated pom

File: bin/vscode-setup [hunk 2]
```diff
@@ -25,7 +25,11 @@ rm -rf server/temp
 
 BUILD=$(
          cat <<END
-<build>
+<properties>
+        <maven.compiler.source>11</maven.compiler.source>
+        <maven.compiler.target>11</maven.compiler.target>
+    </properties>
+    <build>
```
[the remark this answers is not in the document]
Comment (clouser2) [reply bin/vscode-setup:32]: Done. This was originally dedented since the perl replacement below didn't account for leading whitespace. I've updated that replacement so that it also captures leading whitespace for the line.

File: server/build.sbt (hunk 1 above)
[the remark this answers is not in the document]
Comment (clouser2) [reply server/build.sbt:70]: Agreed that we could modify the generated pom. However, I'd prefer to do as little hand-editing as we can. The `auto-value-parent` package doesn't ever have a JAR, so this change should be a no-op. That said, good call on manually testing.

I confirmed that `bin/run-dev` continues to load. Additionally, I ran the docker build from the [build and push script](https://github.com/seattle-uat/civiform/blob/cc9628a5f398c5d19720f8acdb06bdadf90ab036/bin/build-and-push#L5-L17). I didn't think it necessary to push an artifact to DockerHub since the build itself is what we're verifying didn't change. Finally, passing browser / unit test should hopefully bolster confidence. Updated the "Tested" section of the PR to reflect this.

File: .github/workflows/scorecards.yml [hunk 3]
```diff
@@ -22,12 +22,12 @@ jobs:
 
     steps:
       - name: "Checkout code"
-        uses: actions/checkout@a12a3943b4bdde767164f792f33f40b04645d846 # v3.0.0
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
```
Comment (kmcnellis) [inline .github/workflows/scorecards.yml:25]: I don't think this is related

File: README.md [hunk 4]
```diff
@@ -1,4 +1,5 @@
 ![CI status](https://github.com/seattle-uat/civiform/actions/workflows/push.yaml/badge.svg)
+[![CII Best Practices](https://bestpractices.coreinfrastructure.org/projects/6008/badge)](https://bestpractices.coreinfrastructure.org/projects/6008)
```
Comment (kmcnellis) [inline README.md:2]: This is probably a separate change

File: browser-test/package.json [hunk 5]
```diff
@@ -8,7 +8,7 @@
   },
   "devDependencies": {
     "@types/jest": "27.4.1",
-    "@types/node": "16.11.27",
+    "@types/node": "16.11.31",
```
Comment (kmcnellis) [inline browser-test/package.json:11]: Unrelated

File: server/build.sbt [hunk 6]
```diff
@@ -22,7 +22,7 @@ lazy val root = (project in file("."))
       "com.j2html" % "j2html" % "1.4.0",
 
       // Amazon AWS SDK
-      "software.amazon.awssdk" % "aws-sdk-java" % "2.17.161",
+      "software.amazon.awssdk" % "aws-sdk-java" % "2.17.179",
```
Comment (kmcnellis) [inline server/build.sbt:25]: Unrelated

File: server/build.sbt (hunk 1 above)
[the remark this answers is not in the document]
Comment (kmcnellis) [reply server/build.sbt:70]: Awesome!

File: server/package.json [hunk 7]
```diff
@@ -10,7 +10,7 @@
   },
   "license": "CC0-1.0",
   "devDependencies": {
-    "@tailwindcss/line-clamp": "0.3.1",
+    "@tailwindcss/line-clamp": "0.4.0",
```
Comment (kmcnellis) [inline server/package.json:13]: Unrelated

**Review by kmcnellis — APPROVED** [review]

**Review by kmcnellis — APPROVED** [review]

**Review by gwendolyngoetz — APPROVED** [review]
